src/main/java/com/opencart/steps/SearchProductPageBL.java

```diff
@@ -25,26 +28,61 @@ public SearchProductPageBL makeSearch(String value) {
         searchProductPage.getInputSearchCriteria().clear();
         searchProductPage.getInputSearchCriteria().sendKeys(value);
         clickOnSearchGroupButton();
+        selectCategoryFromDropdown("Desktops");
+        clickOnSearchButton();
         searchProductResultPage = new SearchProductResultPage();
         return this;
     }
 
-    public void verifySearchResult(String searchCriteria) {
-        Assert.assertTrue(findAllProductBySearchCriteria(searchCriteria));
+    public void verifyPositiveSearchResult(String searchCriteria) {
+        SoftAssert softAssert = new SoftAssert();
+        String expectedResult = "Products meeting the search criteria";
+        String actualResult = searchProductResultPage.getSuccessTitle().getText();
+        softAssert.assertEquals(expectedResult, actualResult, "Title is correct, now you can check the list of products");
+        softAssert.assertTrue(findAllProductBySearchCriteria(searchCriteria),
+                "Check the list of products Done!");
+        softAssert.assertAll();
+    }
+
+    public void verifyNegativeSearchResult() {
+        String expectedResult = "There is no product that matches the search criteria.";
+        String actualResult = searchProductResultPage.getNoProductsFoundTitle().getText();
+        Assert.assertEquals(expectedResult, actualResult, "Search criteria is actually correct");
     }
 
     private boolean findAllProductBySearchCriteria(String searchCriteria) {
         List<ProductContainer> productContainers = searchProductResultPage.getProductContainers();
+        boolean isProductListContainsSearchCriteria = true;
         for (ProductContainer productContainer : productContainers) {
             if (!productContainer.getName().contains(searchCriteria)) {
-                return false;
+                isProductListContainsSearchCriteria = false;
             }
         }
-        return true;
+        return isProductListContainsSearchCriteria;
     }
 
     private SearchProductPageBL clickOnSearchGroupButton() {
         searchProductPage.getInputSearchGroupButton().click();
         return this;
     }
+
+// When I use the button from searchProductResultPage
+// I catch NullPointerException in the makeSearch()?
+    private SearchProductPageBL clickOnSearchButton() {
```

Comment:
It happens because you have two search inputs located in different places. First -"*.fa-search" and second ".//input[@value='Search']". In this case, you try to click on the search when it still not appeared

Reply:
Create two methods, makeSearch(String phrase) and makeSearch(String phrase, String category), and realize different logic for those two methods